# 1.Import necessary packages

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# 2.Load the file

In [2]:
data = pd.read_csv('./Inc_Exp_Data.csv')

# 3.Analyze the data

In [6]:
data.head()

Mthly_HH_Income  Mthly_HH_Expense  No_of_Fly_Members  Emi_or_Rent_Amt  \
0             5000              8000                  3             2000   
1             6000              7000                  2             3000   
2            10000              4500                  2                0   
3            10000              2000                  1                0   
4            12500             12000                  2             3000   

   Annual_HH_Income Highest_Qualified_Member  No_of_Earning_Members  
0             64200           Under-Graduate                      1  
1             79920               Illiterate                      1  
2            112800           Under-Graduate                      1  
3             97200               Illiterate                      1  
4            147000                 Graduate                      1

In [8]:
data.describe()

Mthly_HH_Income  Mthly_HH_Expense  No_of_Fly_Members  Emi_or_Rent_Amt  \
count        50.000000         50.000000          50.000000        50.000000   
mean      41558.000000      18818.000000           4.060000      3060.000000   
std       26097.908979      12090.216824           1.517382      6241.434948   
min        5000.000000       2000.000000           1.000000         0.000000   
25%       23550.000000      10000.000000           3.000000         0.000000   
50%       35000.000000      15500.000000           4.000000         0.000000   
75%       50375.000000      25000.000000           5.000000      3500.000000   
max      100000.000000      50000.000000           7.000000     35000.000000   

       Annual_HH_Income  No_of_Earning_Members  
count      5.000000e+01              50.000000  
mean       4.900190e+05               1.460000  
std        3.201358e+05               0.734291  
min        6.420000e+04               1.000000  
25%        2.587500e+05               1.000000  
50%        4.474200e+05               1.000000  
75%        5.947200e+05               2.000000  
max        1.404000e+06               4.000000

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 7 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Mthly_HH_Income           50 non-null     int64 
 1   Mthly_HH_Expense          50 non-null     int64 
 2   No_of_Fly_Members         50 non-null     int64 
 3   Emi_or_Rent_Amt           50 non-null     int64 
 4   Annual_HH_Income          50 non-null     int64 
 5   Highest_Qualified_Member  50 non-null     object
 6   No_of_Earning_Members     50 non-null     int64 
dtypes: int64(6), object(1)
memory usage: 2.9+ KB


# 4.What is the Mean Expense of a Household?

In [10]:
data.Mthly_HH_Expense.mean()

18818.0

# 5.What is the Median Household Expense?

In [11]:
data.Mthly_HH_Expense.median()

15500.0

# 6.What is the Monthly Expense for most of the Households?

In [13]:
data.Mthly_HH_Expense.mode()

0    25000
dtype: int64

In [23]:
mth_exp_tmp = pd.crosstab(index=data["Mthly_HH_Expense"], columns="count")
mth_exp_tmp

col_0             count
Mthly_HH_Expense       
2000                  1
4500                  1
5000                  1
6600                  1
7000                  1
8000                  3
9000                  3
10000                 5
10500                 1
12000                 3
12300                 1
13000                 1
15000                 3
16000                 1
18000                 1
19000                 1
20000                 6
22000                 1
25000                 8
30000                 1
40000                 2
45000                 1
48000                 1
50000                 2

In [24]:
mth_exp_tmp.reset_index(inplace=True)
mth_exp_tmp

col_0  Mthly_HH_Expense  count
0                  2000      1
1                  4500      1
2                  5000      1
3                  6600      1
4                  7000      1
5                  8000      3
6                  9000      3
7                 10000      5
8                 10500      1
9                 12000      3
10                12300      1
11                13000      1
12                15000      3
13                16000      1
14                18000      1
15                19000      1
16                20000      6
17                22000      1
18                25000      8
19                30000      1
20                40000      2
21                45000      1
22                48000      1
23                50000      2

In [28]:
mth_exp_tmp[mth_exp_tmp['count'] == data.Mthly_HH_Expense.value_counts().max()]

col_0  Mthly_HH_Expense  count
18                25000      8

# 7.Plot the Histogram to count the Highest qualified member

# 8.Calculate IQR(difference between 75% and 25% quartile)

# 9.Calculate Standard Deviation for first 4 columns.

# 10.Calculate Variance for first 3 columns.

# 11.Calculate the count of Highest qualified member.

# 12.Plot the Histogram to count the No_of_Earning_Members

# 13.Suppose you have option to invest in Stock A or Stock B. The stocks • have different expected returns and standard deviations. The expected return of Stock A is 15% and Stock B is 10%. Standard Deviation of the returns of these stocks is 10% and 5% respectively.

Which is better investment?